service/history/workflow/context.go

```diff
@@ -915,13 +915,20 @@ func (c *ContextImpl) ReapplyEvents(
 
 // TODO: remove `ms` parameter again (added since it's not possible to initialize a new Context with a specific MutableState)
 func (c *ContextImpl) UpdateRegistry(ctx context.Context, ms MutableState) update.Registry {
+	var ns *namespace.Namespace
+	if c.MutableState != nil {
+		ns = c.MutableState.GetNamespaceEntry()
+	} else {
+		ns = ms.GetNamespaceEntry()
+	}
+
+	if c.updateRegistry != nil && c.updateRegistry.FailoverVersion() != ns.FailoverVersion() {
```

Comment:
At least now, the GetCurrentVersion from mutable state will stop updating after workflow is closed, while ns.FailoverVersion will keep changing.  I think that's probably not something you want?

Reply:
I definitely don't want to compare fields which are always different after WF is closed :-) I will use namespace version for both.